pkg/ottl/ottlfuncs/func_uri.go

```diff
@@ -0,0 +1,93 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type URIArguments[K any] struct {
+	URI ottl.StringGetter[K]
+}
+
+func NewURIFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("URI", &URIArguments[K]{}, createURIFunction[K])
+}
+
+func createURIFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*URIArguments[K])
+	if !ok {
+		return nil, fmt.Errorf("URIFactory args must be of type *URIArguments[K]")
+	}
+
+	return URI(args.URI)
+}
+
+func URI[K any](uriSource ottl.StringGetter[K]) (ottl.ExprFunc[K], error) {
```

Comment:
We don't want to export this function so it needs to be named `uri`

Reply:
this is tricky, linter complains when it's lowercase :D 
i can add linter override

Reply:
lol it complains about `uri`?